.gitignore

```diff
@@ -1,26 +1,47 @@
-*.pyc
-*~
+# Byte-compiled / optimized / DLL files
+*.py[cd]
+
+# C extensions
 *.so
-*.pyd
-.*.swp
+
+# Distribution / packaging
+build/
+dist/
+*.egg
 *.egg-info
-.coverage
-.cache
-.env
 MANIFEST
+
+# Unit test / coverage reports
+.cache
+.coverage
+coverage.xml
+htmlcov
 .pytest_cache/
-.ipynb_checkpoints/
-.vscode/
+results/
 result_images/
 tmp-test-dir-with-unique-name/
-coverage.xml
-htmlcov
-build/
-dist/
+
+# Sphinx documentation
+doc/api/gmt*.rst
+doc/api/generated
 doc/_build
-doc/modules
 doc/gallery
-doc/tutorials
+doc/modules
 doc/projections
-doc/api/gmt*.rst
-doc/api/generated
+doc/tutorials
+
+# Jupyter Notebook
+.ipynb_checkpoints/
```

Comment:
Can probably remove this since we don't use Jupyter notebooks anymore. `*.ipynb` files were converted to `*.py` as of #268

```suggestion
```